src/backend/mpi/swdsm.cpp

```diff
@@ -1253,3 +1201,19 @@ size_t argo_get_global_size(){return size_of_all;}
 unsigned long get_classification_index(uint64_t addr){
 	return (2*(addr/(pagesize*CACHELINE))) % classificationSize;
 }
+
+bool _is_cached(std::size_t addr) {
+	argo::node_id_t homenode;
+	std::size_t aligned_address = align_backwards(
+			addr-reinterpret_cast<std::size_t>(startAddr), CACHELINE*pagesize);
+	if(dd::is_first_touch_policy()){
+		homenode = peek_homenode(aligned_address);
+	}else{
+		homenode = get_homenode(aligned_address);
+	}
```

Comment:
if we are already doing virtual lookups for `data_distribution` I do wonder why this branch is needed, but this is probably beyond this PR to fix.